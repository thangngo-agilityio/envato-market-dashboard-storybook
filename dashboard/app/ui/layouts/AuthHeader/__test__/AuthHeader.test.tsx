import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

// component
import { AuthHeader } from '../..';

const mockProps = {
  title: 'Something',
};

describe('AuthHeader component render', () => {
  it('Should render match with snapshot.', () => {
    const renderComponent = () => render(<AuthHeader {...mockProps} />);
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });

  it('renders additional text when pathName is not /forgot-password', () => {
    const { getByText } = render(
      <AuthHeader {...mockProps} pathName="/some/path" />,
    );

    // Assert that the additional text is rendered
    const additionalTextElement = getByText('Send, spend and save smarter', {
      exact: false,
    });
    expect(additionalTextElement).toBeInTheDocument();
  });

  it('does not render additional text when pathName is /forgot-password', () => {
    const { queryByText } = render(
      <AuthHeader {...mockProps} pathName="/forgot-password" />,
    );

    // Assert that the additional text is not rendered with /forgot-password pathName
    const additionalTextElement = queryByText('Send, spend and save smarter', {
      exact: false,
    });
    expect(additionalTextElement).toBeNull();
  });
});
