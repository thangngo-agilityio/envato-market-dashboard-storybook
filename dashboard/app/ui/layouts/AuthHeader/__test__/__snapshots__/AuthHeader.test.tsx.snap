// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader component render Should render match with snapshot. 1`] = `
<div>
  <header
    class="chakra-stack css-tl3ftk"
  >
    <h1
      class="chakra-heading css-13zsi6e"
    >
      Something
    </h1>
    <p
      class="chakra-text css-1xbgeyr"
    >
      Send, spend and save smarter
    </p>
  </header>
</div>
`;
